### DISCUSSION TITLE
finksql不支持on duplicate key

### DISCUSSION THREAD

**Discussion by xmRobinYu** [apache/flink-cdc#1343] [Q&A]
我在使用flink-cdc同步数据的时候碰到一个问题，需求是同步到的目标表采用自增id，需要根据目标表的唯一性字段，更新某些列的数据或者新增数据。我用mysql的insert into ...on duplicate key update能满足改需求，但是发现目前flink sql不支持该语法，想请教大佬们有什么比较好的方案推荐？

**Comment by chem2code** [comment]
源表支持cdc模式就可以了，sink表用复合主键不用自增id
